routes/admin-login.js

```diff
@@ -1,8 +1,54 @@
 var express = require('express');
 var router = express.Router();
+var firebase = require('firebase');
+var config = require('./config.js');
+
+if(!firebase.apps.length)
+	firebase.initializeApp(config.config);
 
 router.get('/', function(req, res, next) {
 	res.render('admin-login');
 });
 
+router.post('/', signIn, function(req, res) {
+	//checks if user is signed in
+	if(firebase.auth().currentUser)
+		res.jsonp({success: true});
+
+	//sign in unsuccessful
+	else
+		res.jsonp({success: false});
+
+});
+
+//this method gets called before function body inside router.post, attemps to sign in user
+function signIn(req, res, next){
+	const email = req.body.email
+	const password = req.body.password;
+
+	firebase.auth().signInWithEmailAndPassword(email, password).catch(function(error) {
+		//sign in unsuccessful
+		console.log("invalid credential")
+	});
+
+	//set 1.2 second timeout so that above method can finish before moving on
+	setTimeout(next, 1200);
+}
+
+//checks if signed in for the navbar 
+router.post('/checkIfSignedIn', function(req, res){
```

Comment:
Should be a GET and should check if the cookie's in the request. A bit hacky and I think the usual way to do this is to set a second cookie that's more insecure but acts as a flag for if the user is authenticated or not.